swingset/src/main/java/com/nqadmin/swingset/SSDataNavigator.java

```diff
@@ -778,20 +814,30 @@ public void actionPerformed(final ActionEvent ae) {
 		// NOTHING HAPPENS.
 		txtCurrentRow.addKeyListener(new KeyAdapter() {
 			@Override
-			public void keyReleased(final KeyEvent ke) {
-				logger.debug("Record number manually updated.");
-				removeRowsetListener();
+			public void keyPressed(final KeyEvent ke) {
+				
 				if (ke.getKeyCode() == KeyEvent.VK_ENTER) {
+					
 					try {
+						removeRowsetListener();
+						
+						if (!commitChanges()) return;
+						
```

Comment:
I see the keyReleased to keyPressed. And this commitChanges().

I know I include minor/cosmetic changes with bug fixes together in the same commit. I shall endeavor to do that no more; it makes it tricky to get to the meat of it.